### PR TITLE
Use UID and GID for USER in Dockerfile

### LINKED ISSUE #1921: Zarf can't init when nonroot user is strictly enforced
### Environment
Device and OS: Ubuntu/WSL
App version: 0.28.2
Kubernetes distro being used: RKE2
Other: Admission controller will not permit containers with named USER because it may be set to 0 in image.

### Steps to reproduce
1. Setup a kubernetes cluster which enforces non-root by default including blocking named users.
2. run `zarf init`

### Expected result
Zarf inits

### Actual Result
zarf agent deployment fails because the user ID can't be looked up before the container starts. 

### Visual Proof (screenshots, videos, text, etc)
image
image

Relevant code:
https://github.com/defenseunicorns/zarf/blob/main/Dockerfile#L4

I'll submit an PR for this shortly...

### PR DESCRIPTION
**Pull request by flickerfly**
## Description

On clusters that strictly enforce no root containers via an Admission Controller, they can't determine that a named user isn't 0 in the container. This changes the container to identify the USER by UID and GID so the admission controller can allow this through.

Chainguard documents the UID and GID of nonroot https://edu.chainguard.dev/chainguard/chainguard-images/reference/static/overview/#users

## Related Issue

Fixes #1921

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: Dockerfile
Comment (Racer159) [reply Dockerfile:6]: Thanks for the PR! Just for readability would request the `chown` be made the same UID with a comment left explaining what the UID refers to, otherwise LGTM.

**Review by Racer159 — CHANGES_REQUESTED** [review]

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by YrrepNoj — APPROVED** [review]

**Review by Racer159 — APPROVED** [review]
lgtm (approving after main merge)

**Review by YrrepNoj — APPROVED** [review]

**Comment by flickerfly** [comment]
Thank you! If I understand right, looks like you folks as the vendor are also owners on the Repo1 equivalent so I'll make the connection.

https://repo1.dso.mil/dsop/opensource/defenseunicorns/zarf/zarf-agent/-/issues/39

**Pull request merged**
